packages/query-core/src/tests/utils.ts

```diff
@@ -6,7 +6,7 @@ import * as utils from '../utils'
 
 export function createQueryClient(config?: QueryClientConfig): QueryClient {
   jest.spyOn(console, 'error').mockImplementation(() => undefined)
```

Comment:
this needs to be removed from here I thinks

Reply:
Removing this somehow leads to consoleMock spy to not work, after removing this I had to add this in every test file to stop consoles from spilling while testing, not sure if it is the correct way,
```tsx
let consoleMock : jest.SpyInstance

  beforeEach(() => {
    consoleMock = jest.spyOn(console, 'error').mockImplementation(() => undefined)
  })

  afterEach(() => {
    consoleMock.mockRestore()
  })
```

not adding this results in console errors to be logged even if the tests are passing.

Reply:
also not sure that this is correct. Tests shouldn't generally write to the console, so what did show up on the console without mocking it?

There are some tests that explicitly do something wrong where an error is expected where we do need to mock the console, but those should be the minority.

This implies that you remove the general error logging, otherwise, every erroneous query spams the console, and we do test error states quite a bit :)

Reply:
yep, it's still here. That's the reason:

<img width="883" alt="Screenshot 2022-12-30 at 11 02 11" src="https://user-images.githubusercontent.com/1021430/210058205-4ae9e334-7245-4630-be64-672d6ee48930.png">

Reply:
also for this test, we'd probably need to mock `console.warn` now:

<img width="1086" alt="Screenshot 2022-12-30 at 11 04 11" src="https://user-images.githubusercontent.com/1021430/210058375-f095806e-6cf0-4912-b9fe-0a92acb85bbf.png">

we don't need to worry about the tests that log further deprecation warnings, because those warnings will be removed when we remove the deprecated features, which is also planned for v5. So now we are in the situation where multiple features are worked on at the same time that kind of intersect 😅 . Maybe we should wait with merging the logger-removal feature until all the other features that use the logger for deprecation warnings have been merged.

Reply:
Good call, as most of the consoles are for deprecations, until then I will dive a little more in the source to get better understanding of things :)

I will try removing the general error logging tonight and try again, from the above screenshot I guess that it is referncing to the query.ts 

```tsx
 if (!isCancelledError(error)) {
        // Notify cache callback
        this.cache.config.onError?.(error, this as Query<any, any, any, any>)

        if (process.env.NODE_ENV !== 'production') {
           console.error(error)
         }
      }
 ```
 and mutation.ts 
 
 ```tsx
  if (process.env.NODE_ENV !== 'production') {
          console.error(error)
        }
  ```
  
  also found out that some tests does not explicitly call console in them but the function they are using calls a console internally and that skips the mock implementation and gets logged to the console. like this test is queryObserver.test.tsx
  
  ```tsx
  test('should prefer isDataEqual to structuralSharing', async () =>{}
  ```
  
  wrapping this test in mockConsole solves the error being printed to the console as the test is passing.